src/main/java/uk/nhs/tis/trainee/usermanagement/api/UserAccountResource.java

```diff
@@ -129,4 +136,66 @@ ResponseEntity<Void> deleteCognitoAccount(@PathVariable String username) {
     log.info("Deleted Cognito account for user '{}'.", username);
     return ResponseEntity.noContent().build();
   }
+
+  /**
+   * Add the given user into Cognito user group.
+   *
+   * @param usergroup The name of the Cognito user group.
+   * @param username The username of the user.
+   * @return 204 No Content, if successful.
+   */
+  @PostMapping("/add-usergroup")
+  ResponseEntity<Void> addToUserGroup(
+      @RequestParam(value = "usergroup", required = true) String usergroup,
+      @RequestParam(value = "username", required = true) String username
+  ) {
+    String groupName;
+    log.info("User '{}' enrolment to user group '{}' requested.", username, usergroup);
+    if (usergroup.equals(DSP_CONSULTATION)) {
+      groupName = consultationGroupName;
+    } else {
+      log.warn("Invalid user group: '{}'", usergroup);
+      return ResponseEntity.badRequest().build();
+    }
+
+    AdminAddUserToGroupRequest request = new AdminAddUserToGroupRequest();
+    request.setUserPoolId(userPoolId);
+    request.setGroupName(groupName);
+    request.setUsername(username);
+
+    cognitoIdp.adminAddUserToGroup(request);
+    log.info("User '{}' is enroled in user group name '{}'.", username, groupName);
+    return ResponseEntity.noContent().build();
+  }
+
+  /**
+   * Remove the given user from Cognito user group.
+   *
+   * @param usergroup The name of the Cognito user group.
+   * @param username The username of the user.
+   * @return 204 No Content, if successful.
+   */
+  @PostMapping("/remove-usergroup")
+  ResponseEntity<Void> removeFromUserGroup(
+      @RequestParam(value = "usergroup", required = true) String usergroup,
+      @RequestParam(value = "username", required = true) String username
+  ) {
```

Comment:
We need to keep this fairly well locked down to restrict what groups can be interacted with, so there is no need for a generic endpoint for all user groups. We should instead just accept a username to an endpoint for just the consultation group.

Reply:
Updated.